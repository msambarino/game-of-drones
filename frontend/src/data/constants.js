export const DEFAULT_RULES = {
  moves: [
    { move: 'rock', kills: 'scissors' },
    { move: 'paper', kills: 'rock' },
    { move: 'scissors', kills: 'paper' }
  ],
  roundWinsRequired: 3
}

export const ROUND_RESULTS = {
  DRAW: 'DRAW',
  PLAYER_1_WINS: 'PLAYER_1_WINS',
  PLAYER_2_WINS: 'PLAYER_2_WINS'
}
