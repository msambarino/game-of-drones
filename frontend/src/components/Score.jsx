import React from 'react'
import PropTypes from 'prop-types'
import { ROUND_RESULTS } from 'data/constants'

const formatRoundWinner = (player1, player2, result) => {
  switch (result) {
    case ROUND_RESULTS.DRAW:
      return '--'
    case ROUND_RESULTS.PLAYER_1_WINS:
      return player1
    case ROUND_RESULTS.PLAYER_2_WINS:
      return player2
    default:
      return null
  }
}

const Score = ({ rounds, player1, player2 }) => (
  <table className="table is-hoverable">
    <thead>
      <tr>
        <th>Round</th>
        <th>{`${player1}'s move`}</th>
        <th>{`${player2}'s move`}</th>
        <th>Winner</th>
      </tr>
    </thead>
    <tbody>
      {rounds.map((round, index) => (
        <tr key={index}>
          <td>{index + 1}</td>
          <td>{round.player1Move}</td>
          <td>{round.player2Move}</td>
          <td>{formatRoundWinner(player1, player2, round.result)}</td>
        </tr>
      ))}
    </tbody>
  </table>
)

Score.propTypes = {
  rounds: PropTypes.array.isRequired,
  player1: PropTypes.string.isRequired,
  player2: PropTypes.string.isRequired
}

export default Score
