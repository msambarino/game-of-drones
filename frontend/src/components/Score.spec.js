import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { ROUND_RESULTS, DEFAULT_RULES } from 'data/constants'
import Score from './Score'

test('snapshot test', () => {
  const move = DEFAULT_RULES.moves[0].move
  const rounds = [
    { result: ROUND_RESULTS.DRAW, player1Move: move, player2Move: move },
    { result: ROUND_RESULTS.DRAW, player1Move: move, player2Move: move },
    { result: ROUND_RESULTS.DRAW, player1Move: move, player2Move: move }
  ]

  const component = shallow(<Score rounds={rounds} player1="p1" player2="p2" />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('renders a row for each round', () => {
  const move = DEFAULT_RULES.moves[0].move
  const rounds = [
    { result: ROUND_RESULTS.DRAW, player1Move: move, player2Move: move },
    { result: ROUND_RESULTS.DRAW, player1Move: move, player2Move: move },
    { result: ROUND_RESULTS.DRAW, player1Move: move, player2Move: move }
  ]

  const component = shallow(<Score rounds={rounds} player1="p1" player2="p2" />)

  expect(component.find('tbody > tr')).toHaveLength(3)
})
