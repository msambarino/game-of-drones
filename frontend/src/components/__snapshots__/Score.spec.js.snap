// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<table
  className="table is-hoverable"
>
  <thead>
    <tr>
      <th>
        Round
      </th>
      <th>
        p1's move
      </th>
      <th>
        p2's move
      </th>
      <th>
        Winner
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        1
      </td>
      <td>
        rock
      </td>
      <td>
        rock
      </td>
      <td>
        --
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        2
      </td>
      <td>
        rock
      </td>
      <td>
        rock
      </td>
      <td>
        --
      </td>
    </tr>
    <tr
      key="2"
    >
      <td>
        3
      </td>
      <td>
        rock
      </td>
      <td>
        rock
      </td>
      <td>
        --
      </td>
    </tr>
  </tbody>
</table>
`;
